### ISSUE TITLE
Serde `flatten` does not preserve order

### ISSUE THREAD

**Issue by ralpha** [GREsau/schemars#122]
When you have a struct with the following structure:
```rust
#[derive(Serialize, Deserialize, JsonSchema)]
pub struct MyStruct {
    #[serde(flatten)]
    pub item: OtherItem,
    pub my_int: i32,
}

#[derive(Serialize, Deserialize, JsonSchema)]
pub struct OtherItem {
    pub something: i32,
}
```

This will result in an schema that describes this will be returned:
```json
{
   "my_int": 0,
   "something": 0,
}
```
Even tho I expect (and serde returns):
```json
{
   "something": 0,
   "my_int": 0,
}
```

This is probably because of the way Schemars handles the flatten part: https://github.com/GREsau/schemars/blob/master/schemars_derive/src/schema_exprs.rs#L565:L566

They should instead be inserted in the correct place. But looking at the code this requires a slightly different approach of inserting them into the object schema.

They are now split apart from `properties` here: https://github.com/GREsau/schemars/blob/e9d5f4057e84f37a4519a5413ddded82cb13ea71/schemars_derive/src/schema_exprs.rs#L470
Instead they should be left together and have an `if` catch for when looping over the fields (in [the `map`](https://github.com/GREsau/schemars/blob/e9d5f4057e84f37a4519a5413ddded82cb13ea71/schemars_derive/src/schema_exprs.rs#L480)).

**Comment by strohel** [comment]
Hi @GREsau, I've faced this issue recently. I see there's a PR by @SamirTalwar that says it fixes it. Could you please take a look?

@GREsau/@SamirTalwar, I can help with rebasing the PR on top of latest `master` to fix conflicts if desired.

**Comment by SamirTalwar** [comment]
I’m happy to update the pull request myself if there’s a sign that it will be considered for merging.

**Comment by strohel** [comment]
> I’m happy to update the pull request myself if there’s a sign that it will be considered for merging.

BTW, thanks for the patch @SamirTalwar! Tt fixed the last blocker for using `schemars` for us and we're now using your patched version. In that sense we'd already benefit from you rebasing it on latest `master`, though agreed upstreaming it would be the best. :pray:

**Comment by SamirTalwar** [comment]
@strohel, I have rebased the PR. Hope it helps!

**Comment by GREsau** [comment]
This is fixed in [schemars 1.0.0-alpha.3](https://crates.io/crates/schemars/1.0.0-alpha.3) (I think it was actually fixed in alpha.1!)
